cpu/stm32f0/periph/gpio.c

```diff
@@ -24,309 +24,96 @@
 #include "periph/gpio.h"
 #include "periph_conf.h"
 
+/**
+ * @brief   Allocate memory for one callback and argument per EXTI channel
+ */
 static gpio_isr_ctx_t gpio_config[GPIO_NUMOF];
```

Comment:
`GPIO_NUMOF` is absolutely wrong here, this is a legacy value that defaults to `0` -> hence leading to not working external interrupts...

Change the top of this file to
```c
/**
 * @brief   The STM32F0 family has 16 external interrupt lines
 */
#define EXTI_NUMOF          (16U)

/**
 * @brief   Allocate memory for one callback and argument per EXTI channel
 */
static gpio_isr_ctx_t gpio_config[EXTI_NUMOF];
```
and see below on how to change the isr